### PR TITLE
Traefik

### PR DESCRIPTION
**Pull request by mira-miracoli**
Add documentation for traefik

### CODE REVIEW

File: traefik.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+- Go to [aws](https://signin.aws.amazon.com/) and sign in using the credentials in the vault (`aws.yml`)
+- Navigate to `route53`
+- click on `hosted zones`
+- then on `usegalaxy.eu`
+- change the `A Record` for `usegalaxy.eu` and point it to sn06's public IP address (`132.230.223.239`)
+- The record usually has a TTL of 7200s, which means, that after 2h all requests should get sn06's IP instead of Traefik's.
+- In order to bridge that time, you can install nginx on Traefik and `proxy_pass` all requests to one headnode directly.
+### usegalaxy.eu /subdomain is showing a plain `404 not found`
+Most likely something happened to the router.
+- Check the [dashboard](https://traefik./gspringhare-dinosaur.ts.net/dashboard/#/) via tailscale
```
Comment (kysrpex) [inline traefik.md:45]: Typo?

```suggestion
- Check the [dashboard](https://traefik.gspringhare-dinosaur.ts.net/dashboard/#/) via tailscale
```

**Review by kysrpex — COMMENTED** [review]
Summarizing: this documentation is f***ing awesome.

File: traefik.md (hunk 1 above)
Comment (mira-miracoli) [reply traefik.md:45]: yes, a result of editing too much with vim

File: traefik.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+`docker service logs` would not work with Traefik, because it writes directly to `/var/log/traefik`  
+
+
+## New subdomains
+To add new subdomains, add a new line to the file `files/traefik/rules/template-subdomains.yml` in the [infrastructure-playbook](https://github.com/usegalaxy-eu/infrastructure-playbook) repo. The language there is a `go` template for a `yaml` file, which might look similar to ansible at first. (In case you wonder about the syntax).  
+The line should look like the ones above, like this scheme:
+~~~
+{{template "subdomain" "<your-subdomain-name>"}}
+~~~
+Be careful, the word `subdomain` in the second colum needs to stay literaly "subdomain", only the 3rd column is changed to the new subdomain,but without any `.usegalaxy.eu`.
```
Comment (bgruening) [inline traefik.md:30]:
```suggestion
Be careful, the word `subdomain` in the second colum needs to stay literaly "subdomain", only the 3rd column is changed to the new subdomain, but without any `.usegalaxy.eu`.  
```

File: traefik.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+- The record usually has a TTL of 7200s, which means, that after 2h all requests should get sn06's IP instead of Traefik's.
+- In order to bridge that time, you can install nginx on Traefik and `proxy_pass` all requests to one headnode directly.
+### usegalaxy.eu /subdomain is showing a plain `404 not found`
+Most likely something happened to the router.
+- Check the [dashboard](https://traefik.gspringhare-dinosaur.ts.net/dashboard/#/) via tailscale
+- If all routers look fine, check if something happened to the rulefile in `files/traefik/rules/` `usegalaxy-eu-router.yml` for usegalaxy.eu and `template-subdomains.yml` for subdomains. Take a close look at the `HostRegexp` rule.
+- Less likely: check that the `servers` in `usegalaxy-eu-service.yml` are correct and reachable.
+### `Bad Gateway` error
+- One headnode unhealthy? You can test with e.g. `sn07.galaxyproject.eu` directly. Traefik should automatically skip unhealth hosts, see the [dashboard](https://traefik./gspringhare-dinosaur.ts.net/dashboard/#/) and [docs](https://doc.traefik.io/traefik/v3.0/routing/services/#health-check)
```
Comment (bgruening) [inline traefik.md:49]:
```suggestion
- One headnode unhealthy? You can test with e.g. `sn07.galaxyproject.eu` directly. Traefik should automatically skip unhealth hosts, see the [dashboard](https://traefik./gspringhare-dinosaur.ts.net/dashboard/#/) and [docs](https://doc.traefik.io/traefik/v3.0/routing/services/#health-check)
```
```suggestion
- One headnode unhealthy? You can test with e.g. `sn07.galaxyproject.eu` directly. Traefik should automatically skip unhealth hosts, see the [dashboard](https://traefik.springhare-dinosaur.ts.net/dashboard/#/) and [docs](https://doc.traefik.io/traefik/v3.0/routing/services/#health-check)
```

File: traefik.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+- Go to [aws](https://signin.aws.amazon.com/) and sign in using the credentials in the vault (`aws.yml`)
+- Navigate to `route53`
+- click on `hosted zones`
+- then on `usegalaxy.eu`
+- change the `A Record` for `usegalaxy.eu` and point it to sn06's public IP address (`132.230.223.239`)
+- The record usually has a TTL of 7200s, which means, that after 2h all requests should get sn06's IP instead of Traefik's.
+- In order to bridge that time, you can install nginx on Traefik and `proxy_pass` all requests to one headnode directly.
+### usegalaxy.eu /subdomain is showing a plain `404 not found`
+Most likely something happened to the router.
+- Check the [dashboard](https://traefik.gspringhare-dinosaur.ts.net/dashboard/#/) via tailscale
```
Comment (bgruening) [inline traefik.md:45]:
```suggestion
- Check the [dashboard](https://traefik.springhare-dinosaur.ts.net/dashboard/#/) via tailscale
```

File: traefik.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+## New subdomains
+To add new subdomains, add a new line to the file `files/traefik/rules/template-subdomains.yml` in the [infrastructure-playbook](https://github.com/usegalaxy-eu/infrastructure-playbook) repo. The language there is a `go` template for a `yaml` file, which might look similar to ansible at first. (In case you wonder about the syntax).  
+The line should look like the ones above, like this scheme:
+~~~
+{{template "subdomain" "<your-subdomain-name>"}}
+~~~
+Be careful, the word `subdomain` in the second colum needs to stay literaly "subdomain", only the 3rd column is changed to the new subdomain,but without any `.usegalaxy.eu`.  
+Once this is deployed, Traefik will automatically create a router for it and fetch certificates for the subdomain as well as a wildcard certificate for ITs.  
+If you did everything correctly, the new router appears on Traefik's [dashboard](https://traefik./gspringhare-dinosaur.ts.net/dashboard/#/).
```
Comment (bgruening) [inline traefik.md:32]:
```suggestion
If you did everything correctly, the new router appears on Traefik's [dashboard](https://traefik.springhare-dinosaur.ts.net/dashboard/#/).
```

File: traefik.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+Once this is deployed, Traefik will automatically create a router for it and fetch certificates for the subdomain as well as a wildcard certificate for ITs.  
+If you did everything correctly, the new router appears on Traefik's [dashboard](https://traefik.springhare-dinosaur.ts.net/dashboard/#/).
+
+## How to debug
+### 🚑 Galaxy not reachable
+- Go to [aws](https://signin.aws.amazon.com/) and sign in using the credentials in the vault (`aws.yml`)
+- Navigate to `route53`
+- click on `hosted zones`
+- then on `usegalaxy.eu`
+- change the `A Record` for `usegalaxy.eu` and point it to sn06's public IP address (`132.230.223.239`)
```
Comment (bgruening) [inline traefik.md:40]: Should we also document the public IP from traefik or the hostname?

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Very cool! Thanks!

**Pull request merged**